yarn-project/aztec-nr/aztec/src/avm/context.nr

```diff
@@ -0,0 +1,88 @@
+use crate::avm::env_getters::{
+    get_address,
+    get_storage_address,
+    get_origin,
+    get_sender,
+    get_portal,
+    get_fee_per_l1_gas,
+    get_fee_per_l2_gas,
+    get_fee_per_da_gas,
+    get_chain_id,
+    get_version,
+    get_block_number,
+    get_timestamp,
+    get_is_static_call,
+    get_is_delegate_call,
+};
+
+use dep::protocol_types::address::{
+    AztecAddress,
+    EthAddress,
+};
+
+// Getters that will be converted by the transpiler into their
+// own opcodes
+struct AvmContext {}
+
+// No new function as this struct is entirely static getters
+impl AvmContext {
+    pub fn address() -> AztecAddress {
+        get_address()
+    }
+
+    pub fn storage_address() -> AztecAddress {
+        get_storage_address()
+    }
+
+    pub fn origin() -> AztecAddress {
+        get_origin()
+    }
+
+    pub fn sender() -> AztecAddress {
+        get_sender()
+    }
+
+    pub fn portal() -> EthAddress {
+        get_portal()
+    }
```

Comment:
I know this is matching the AVM opcode and context naming. Should we try and match the interface of the existing PublicContext (which itself is close to the PrivateContext) from the current context.nr? That'd use `this_address`, `this_portal_address`?

We can request that aztec-nr's Contexts change to be a little simpler/less-verbose. We can also consider changing EVM terminology to match the existing aztec-nr contexts if we feel there is a divide.

Reply:
It'd be awesome if we could just add the avm tag/macro to a existing function and it'd just "work" because all of its old context accesses still work. Not sure if that's feasible but it'd be beautiful.

Reply:
yessir this would be the next steps when i make the context, was gunna do so in a stack above